### PR TITLE
fix(university-gateway): Client side fetching programs

### PR DESCRIPTION
**Pull request by Ballioli**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Comment by Ballioli** [comment]
image
Haus

File: apps/web/screens/UniversitySearch/UniversitySearch.tsx [hunk 1]
```diff
@@ -139,14 +143,60 @@ const getActiveNavigationItemTitle = (
...
+  >([])
+  const [query, setQuery] = useState('')
+  const searchTermHasBeenInitialized = useRef(false)
+  const [originalSortedResults, setOriginalSortedList] = useState<any>([])
+  const [filteredResults, setFilteredResults] = useState<Array<any>>([])
+  const [gridView, setGridView] = useState<boolean>(true)
+  const { linkResolver } = useLinkResolver()
+  const [totalPages, setTotalPages] = useState<number>(
+    0,
+    //Math.ceil(data.length / ITEMS_PER_PAGE),
```
[the remark this answers is not in the document]
Comment (Ballioli) [reply apps/web/screens/UniversitySearch/UniversitySearch.tsx]: Yea, gone

File: apps/web/screens/UniversitySearch/UniversitySearch.tsx [hunk 2]
```diff
@@ -139,14 +143,60 @@ const getActiveNavigationItemTitle = (
...
+  const [filters, setFilters] = useState<FilterProps>(
+    JSON.parse(JSON.stringify(initialFilters)),
+  )
+
+  useEffect(() => {
+    if (!loading) {
+      const temp = [
+        ...(data?.universityGatewayPrograms.data as UniversityGatewayProgram[]),
+      ]
+        // .sort((x, y) => (x.nameIs > y.nameIs ? 1 : -1))
```
[the remark this answers is not in the document]
Comment (Ballioli) [reply apps/web/screens/UniversitySearch/UniversitySearch.tsx]: Yes, gone now

**Review by davidkef — APPROVED** [review]

File: apps/web/screens/UniversitySearch/UniversitySearch.tsx [hunk 3]
```diff
@@ -1454,19 +1427,44 @@ UniversitySearch.getProps = async ({ apolloClient, locale, query, res }) => {
...
 
+  const { search } = query
+
+  const namespace = JSON.parse(
+    namespaceResponse?.data?.getNamespace?.fields || '{}',
+  ) as Record<string, string>
+
+  let showPagesFeatureFlag = false
+
+  if (publicRuntimeConfig?.environment === 'prod') {
```
[the remark this answers is not in the document]
Comment (Ballioli) [reply apps/web/screens/UniversitySearch/UniversitySearch.tsx]: Removed

**Review by RunarVestmann — APPROVED** [review]

**Pull request merged**